src/client.rs

```diff
@@ -335,11 +336,16 @@ pub async fn insert_points_batch(
 
             let vectors: Vectors = vectors_map.into();
 
-            points.push(PointStruct::new(
-                point_id,
-                vectors,
-                random_payload(Some(payload_count)),
-            ));
+            let mut payload = random_payload(Some(payload_count));
+
+            if timestamp_payload {
+                payload.insert(
+                    "timestamp",
+                    chrono::Utc::now().to_rfc3339_opts(chrono::SecondsFormat::Millis, true),
+                );
+            }
+
+            points.push(PointStruct::new(point_id, vectors, payload));
```

Comment:
Is it possible to conserve the current behavior of not adding automatically a payload on insertion if the new CLI option is not set?

Reply:
But the behavior is exactly the same as it was before, if the option is not set.

E.g., I've changed this:

```rust
            points.push(PointStruct::new(
                point_id,
                vectors,
                random_payload(Some(payload_count)),
            ));
```

To this:
```rust
            let mut payload = random_payload(Some(payload_count));

            if timestamp_payload {
                payload.insert(
                    "timestamp",
                    chrono::Utc::now().to_rfc3339_opts(chrono::SecondsFormat::Millis, true),
                );
            }

            points.push(PointStruct::new(point_id, vectors, payload));
```

`payload` was initialized to `random_payload(Some(payload_count))` even before my change.

It's just when workload calls `insert_points_batch`, it pass `payload_count: 0`, and (I guess), this produces an empty payload.

Reply:
Thanks a lot for the explanation, I forgot about the zero trick :sweat_smile: